### PR TITLE
024 registration system

### PR DESCRIPTION
**Pull request by MihneaBocaKainos**

### CODE REVIEW

File: src/main/java/com/kainos/ea/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+   public UserController() {userService = new UserService(new UserDao());}
+
+   @POST
+   @Path("/register")
+   @Consumes(MediaType.APPLICATION_JSON)
+   @Produces(MediaType.APPLICATION_JSON)
+   public Response createUser(User user) {
+      try {
+         return Response.ok(userService.registerUser(user)).build();
+      } catch(Exception e) {
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/controller/UserController.java:31]: Exception is too broad. `userService` is returning `SQLException` so better to narrow it to that

File: src/main/java/com/kainos/ea/dao/UserDao.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Objects;
+
+import static com.kainos.ea.util.DatabaseConnector.getConnection;
+
+public class UserDao {
+
+    public Token register(User user) {
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/dao/UserDao.java:17]: Might be worth renaming this method to something like `doRegister`

File: src/main/java/com/kainos/ea/dao/UserDao.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            preparedStatement.setString(2, user.getPassword());
+            preparedStatement.setString(3, user.getRole());
+
+
+            if(preparedStatement.executeUpdate() > 0) {
+                Token token = new Token("");
+                token.generateToken(user);
+
+                return token;
+            } else {
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/dao/UserDao.java:41]: else branch is redundant and may be unwrapped for instance

```
if(preparedStatement.executeUpdate() > 0) {
                Token token = new Token("");
                token.generateToken(user);

                return token;
            }
            return null;
```

File: src/main/java/com/kainos/ea/dao/UserDao.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        try {
+
+            if (user.getPassword() == null) {
+                throw new InvalidPasswordException();
+            }
+            if (!user.getEmail().contains("@")) {
+                throw new InvalidEmailException();
+            }
+
+            Connection c = getConnection();
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/dao/UserDao.java:27]: `c` is not of type char so maybe use a descriptive name for this object i.e `connection` or `conn`

File: src/main/java/com/kainos/ea/dao/UserDao.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+public class UserDao {
+
+    public Token register(User user) {
+        try {
+
+            if (user.getPassword() == null) {
+                throw new InvalidPasswordException();
+            }
+            if (!user.getEmail().contains("@")) {
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/dao/UserDao.java:23]: Can we not validate email at front end?

**Review by hellofadude — APPROVED** [review]

**Review by Ryan-Jones1999 — APPROVED** [review]
Looks good cloned down and works as expected

**Pull request merged**